app/models/spree/variant_decorator.rb

```diff
@@ -118,6 +118,11 @@ def refresh_products_cache
     end
   end
 
+  # Deletes the record, skipping callbacks, but it also refreshes the cache
+  def delete_and_refresh_cache
+    destruction { delete }
```

Comment:
this method name destruction... imo it's really not informative and could be better named but it makes me laugh, could be called implosion instead :-D

Reply:
lol. I was just trying to follow the previous naming but I can improve it of course.